cfme/common/vm.py

```diff
@@ -229,6 +229,10 @@ def find_quadicon(self, from_any_provider=False, use_search=True):
         # todo :refactor this method replace it with vm methods like get_state
         if from_any_provider:
             view = navigate_to(self.parent, 'All')
+        elif from_archived_all:
+            view = navigate_to(self.appliance.collections.infra_vms, 'ArchivedAll')
```

Comment:
Can you assume infra_vms collection here? This is a common VM class, used for cloud instances too.

Larger question - why only implement this for infrastructure, when archved and orphaned vm's exist for cloud too? The lack of a BaseCollection dependent BaseProvider might actually be a blocker here to cleanly implementing this, making it a much larger scope than your intended PR.

Given the future planned MIQ provider support, I'm inclined to allow this PR just implementing the infrastructure side of things, but we'll need some enhancements in the future to support both cloud and infra archived/orphaned nav.

All of that said, I think for this method, the collection needs to be based on the provider type. You should be able to achieve this simply with `view = navigate_to(self.appliance.provider_based_collection(self.provider), 'ArchivedAll')`.  Note if this is called with a cloud provider instance instead of an infra VM, this will result in `NavigationDestinationNotFound`.

Reply:
@mshriver 
- I am agree with you, we should have same test case for cloud instances/vms too. Hence I have added test case for cloud providers.
- As cloud instances can not be orphaned; I have added test case for only archived instances